File: ACM/NiuKe/2021.7.26/i.cpp
/*
 * @Description: file content
 * @Author: NEFU AB_IN
 * @version: 1.0
 * @Date: 2021-07-26 13:20:52
 * @LastEditors: NEFU AB_IN
 * @LastEditTime: 2021-07-26 13:26:56
 */
#include <bits/stdc++.h>
#define lowbit(x) ((x) & -(x))
#define LL                          long long
#define ULL                         unsigned long long
#define SZ(X)                       ((int)(X).size())
#define IOS                         ios::sync_with_stdio(false);cin.tie(0);cout.tie(0);

using namespace std;
const LL N = 200005;
LL n, a[N];
LL cnt[N], tree[N];

void add(LL x, LL d)
{
    while (x <= n)
    {
        tree[x] += d;
        x += lowbit(x);
    }
}
LL sum(LL x)
{
    LL sum = 0;
    while (x > 0)
    {
        sum += tree[x];
        x -= lowbit(x);
    }
    return sum;
}
int main()
{
    IOS;
    cin >> n;
    for (LL i = 1; i <= n; i++)
        cin >> a[i];
    for (LL i = 1; i <= n; i++)
    {
        if (cnt[a[i] + 1])
            a[i]++;
        cnt[a[i]]++;
    }
    LL ans = 0;
    for (LL i = 1; i <= n; i++)
    {
        add(a[i], 1);
        ans += (i - sum(a[i]));
    }
    cout << ans << endl;
    return 0;
}

File: ACM/ACnowcoder/2021.8.7/i.py
'''
Description: file content
Author: NEFU AB_IN
version: 1.0
Date: 2021-08-07 12:10:10
LastEditors: NEFU AB_IN
LastEditTime: 2021-08-07 12:55:37
'''
x, s = list(map(int, input().split()))
x = bin(x)[2:][::-1]
s = bin(s)[2:][::-1]

if len(x) > len(s):
    print(0)
    exit(0)
ans = 1
for i in range(len(x)):
    if x[i] == '1' and s[i] == '0':
        ans = 0
        break
    if x[i] == '1' and s[i] == '1':
        ans *= 2
print(x, s)
print(ans)
